crates/cli/src/commands/order_clear/list.rs

```diff
@@ -0,0 +1,91 @@
+use std::{fs::canonicalize, path::PathBuf};
+
+use crate::{
+    execute::Execute,
+    subgraph::{CliPaginationArgs, CliSubgraphArgs},
+};
+use anyhow::Result;
+use clap::Args;
+use comfy_table::Table;
+use rain_orderbook_common::subgraph::SubgraphArgs;
+use rain_orderbook_subgraph_client::{
+    types::flattened::{OrderClearFlattened, TryIntoFlattenedError},
+    PaginationArgs, WriteCsv,
+};
+use tracing::info;
+
+#[derive(Args, Clone)]
+pub struct CliOrderClearListArgs {
+    #[arg(long, help = "Write results to a CSV file at the path provided")]
+    pub csv_file: Option<PathBuf>,
+
+    #[clap(flatten)]
+    pub pagination_args: CliPaginationArgs,
+
+    #[clap(flatten)]
+    pub subgraph_args: CliSubgraphArgs,
+}
+
+impl Execute for CliOrderClearListArgs {
+    async fn execute(&self) -> Result<()> {
+        let subgraph_args: SubgraphArgs = self.subgraph_args.clone().into();
+        let pagination_args: PaginationArgs = self.pagination_args.clone().into();
+        let order_clears = subgraph_args
+            .to_subgraph_client()
+            .await?
+            .order_clears_list(pagination_args)
+            .await?;
+        let order_clears_flattened: Vec<OrderClearFlattened> = order_clears
+            .into_iter()
+            .map(|o| o.try_into())
+            .collect::<Result<Vec<OrderClearFlattened>, TryIntoFlattenedError>>()?;
+
+        if let Some(csv_file) = self.csv_file.clone() {
```

Comment:
i think we want csv formatting but not file writing, this is a cli so it's easy for the caller to simply pipe the stdout to a file if they want that, but the inverse is more difficult